binlog_streamer.go

```diff
@@ -12,6 +12,7 @@ import (
 	"github.com/go-mysql-org/go-mysql/mysql"
 	"github.com/go-mysql-org/go-mysql/replication"
 	"github.com/sirupsen/logrus"
+	"vitess.io/vitess/go/vt/sqlparser"
```

Comment:
Presumably this is where the +143k lines of code comes from..? Generally speaking every time we do a vendor upgrade, I read through all the changes (and even review some of the code) of the dependencies I'm upgrading (only on the most important ones, like the go-mysql library). Will adding vitess as a dependency make it a lot harder as it introduces a lot of noise?

Are there any other "easy" options to do this parsing?

Reply:
It is no doubt a huge dependency, so I expected this comment. There are some libraries which have extracted the parser (only) out of Vitess, but each of those seems to have stopped development (or alternatively: they are simply complete).

For example https://github.com/blastrain/vitess-sqlparser exposes nearly the same API and is a much smaller dependency.